tests/posix/common/src/pthread_attr.c

```diff
@@ -439,6 +439,114 @@ ZTEST(pthread_attr, test_pthread_attr_setstacksize)
 	}
 }
 
+ZTEST(pthread_attr, test_pthread_attr_getinheritsched)
+{
+	int inheritsched = BIOS_FOOD;
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_getinheritsched(NULL, NULL), EINVAL);
+			zassert_equal(pthread_attr_getinheritsched(NULL, &inheritsched), EINVAL);
+			zassert_equal(pthread_attr_getinheritsched(&uninit_attr, &inheritsched),
+				      EINVAL);
+		}
+		zassert_equal(pthread_attr_getinheritsched(&attr, NULL), EINVAL);
+	}
+
+	zassert_ok(pthread_attr_getinheritsched(&attr, &inheritsched));
+	zassert_equal(inheritsched, PTHREAD_INHERIT_SCHED);
+}
+
+static struct sched_param setinheritsched_sched_param;
+static int setinheritsched_policy;
+static int setinheritsched_inheritsched;
+
+static void *test_pthread_attr_set_inheritsched_child_fn(void *arg)
+{
+	pthread_t self;
+	int policy = SCHED_INVALID;
+	struct sched_param getschedparam = {
+		.sched_priority = 1,
+	};
+
+	self = pthread_self();
+
+	zassert_ok(pthread_getschedparam(self, &policy, &getschedparam));
+
+	if (setinheritsched_inheritsched == PTHREAD_INHERIT_SCHED) {
+		zassert_equal(policy, setinheritsched_policy);
+		zassert_equal(getschedparam.sched_priority,
+				  setinheritsched_sched_param.sched_priority);
+	}
+
+	return NULL;
+}
+
+static void *test_pthread_attr_set_inheritsched_parent_fn(void *arg)
+{
+	pthread_t self;
+
+	self = pthread_self();
+
+	/* Change priority of thread */
+	zassert_ok(pthread_setschedparam(self, setinheritsched_policy,
+				  &setinheritsched_sched_param));
+
+	pthread_t child;
+
+	zassert_ok(pthread_create(&child, NULL,
+					  test_pthread_attr_set_inheritsched_child_fn, NULL));
+	zassert_ok(pthread_join(child, NULL));
+
+	return NULL;
+}
+
+static void test_pthread_attr_set_inheritsched_common(int inheritsched)
+{
+	int getinheritsched = BIOS_FOOD;
+
+	setinheritsched_policy = SCHED_RR;
+	setinheritsched_sched_param.sched_priority = 2;
+	setinheritsched_inheritsched = inheritsched;
+
+	/* Set inheritsched attribute */
+	zassert_ok(pthread_attr_setinheritsched(&attr, inheritsched));
+	zassert_ok(pthread_attr_getinheritsched(&attr, &getinheritsched));
+	zassert_equal(getinheritsched, inheritsched);
+
+	pthread_t parent;
+
+	zassert_ok(pthread_create(&parent, &attr,
+					  test_pthread_attr_set_inheritsched_parent_fn, NULL));
```

Comment:
Does this work, @moonlight83340? With that, it should be possible to compare both the expected and actual values in `test_pthread_attr_set_inheritsched_parent_fn()`
```suggestion
					  test_pthread_attr_set_inheritsched_parent_fn, &attr));
```